prometheus/not_compatible.go

```diff
@@ -63,12 +63,17 @@ func (c *notCompatibleHTTPClient) LabelNames(ctx context.Context, name string) (
 	if err != nil {
 		return nil, err
 	}
-	var result []string
+	// subResult is used as a set of label. Like that we are sure we don't have any duplication
+	subResult := make(map[string]bool)
 	for _, ln := range labelNames {
 		for l := range ln {
-			result = append(result, string(l))
```

Comment:
If deduplication happens here (which is good), the deduplication code in `completion.go` should be deleted.